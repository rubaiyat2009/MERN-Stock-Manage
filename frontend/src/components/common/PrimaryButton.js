import { Button } from '@mui/material';
import React from 'react';

const PrimaryButton = ({ outlined, children, sx, ...props }) => {
    return (
        <Button {...props} sx={{ textTransform: "capitalize", color: "#fff", ...sx }} variant={outlined ? "outlined" : "contained"}>
            {children}
        </Button>
    );
};

export default PrimaryButton;