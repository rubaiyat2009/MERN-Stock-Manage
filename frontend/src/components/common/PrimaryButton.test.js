import PrimaryButton from "./PrimaryButton";
import renderer from "react-test-renderer";

it('with only children props', () => {
    const component = renderer.create(
        <PrimaryButton>Button Test</PrimaryButton>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});

it('with outline props', () => {
    const component = renderer.create(
        <PrimaryButton outlined>Button Test</PrimaryButton>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});

it('with sx props', () => {
    const component = renderer.create(
        <PrimaryButton sx={{ color: "red" }}>Button Test</PrimaryButton>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});