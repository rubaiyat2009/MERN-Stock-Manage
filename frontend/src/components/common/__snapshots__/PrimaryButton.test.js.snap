// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with only children props 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-3u2rsm-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Button Test
</button>
`;

exports[`with outline props 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1r8j2cq-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Button Test
  <span
    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`with sx props 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-ayhyiq-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Button Test
</button>
`;
